### ISSUE TITLE
[Emacs 29.4] Irony not working after emacs restart

### ISSUE THREAD

**Issue by Bibroute** [Sarcasm/irony-mode#593]
Greetings,

Thank you very much for creating this wonderful piece of code that I've been using for years now.

Very recently I have updated emacs to version 29.4, purged and installed a new instance of doom that I am using.
Trouble started since then when I tried to install company-irony again.

When I do install the package, after adding a line in irony.el with `;; Version: 1.6.1`  to make the server installation work, everything's fine.
I can set up irony-cdb-compile-options, with the .clang_complete I've created, and irony-cdb-menu return correct strings.
In a C++ file, autocomplete is done as expected. Though flycheck is stuck with an error on C++ standard library (but that may not be related).

However, after restarting emacs when using M-x commands related to irony have a `(nil)` next to them and, when used, mini-buffer returns: "wrong type argument commandp irony-cdb-compile-options" or "Symbol function definition is void: irony-cdb-autosetup-compile-option".
In short, nothing work anymore. Doing a reinstall of irony resolve the issues but it isn't a problem solver.

In the config.el I copied the code lines stated in the README of this project.
`(add-hook 'c++-mode-hook 'irony-mode)`
`(add-hook 'c-mode-hook 'irony-mode)`
`(add-hook 'objc-mode-hook 'irony-mode)`
`(add-hook 'irony-mode-hook 'irony-cdb-autosetup-compile-options)`
 `(require 'company-irony)`
` (eval-after-load 'company '(add-to-list 'company-backends 'company-irony))`

Every packages are downloaded from elpa.

It would be appreciated if I could have any hint on how to resolve or at least find the source of the issue.
If I forgot any necessary information please ask.

Sincerely,

**Comment by Bibroute** [comment]
company-irony seems to works fine as it can complete standard c++ libraries.
Looking into this issue: [https://github.com/Sarcasm/company-irony/issues/2](https://github.com/Sarcasm/company-irony/issues/2), I did the command C-h v company-backends.
Returned value: Value in buffer [...].C: (company-capf company-yasnippet company-irony)
company-clang is missing.

After adding a line to add company-clang in company-backend, it looks like this:
Value in buffer [...].C: (company-capf company-yasnippet company-irony company-clang)

Though, auto-complete for custom libraries still doesn't work, and irony commands  related to clang still unavailable.

**Comment by Bibroute** [comment]
Problem resolved:
The source was:
- Lacking commands on the config.el
- company-anaconda causing trouble with the rest of the config.

I put here the lines from config.el in case someone have the same trouble:
`(require 'irony)`
`(add-hook 'c++-mode-hook 'irony-mode)`
`(add-hook 'c-mode-hook 'irony-mode)`
`(add-hook 'objc-mode-hook 'irony-mode)`
`(require 'company)`
`(add-hook 'after-init-hook 'global-company-mode)`
`(require 'company-irony)`
`(require 'company-irony-c-headers)`
`(require 'company-clang)`
`(eval-after-load 'company '(add-to-list 'company-backends 'company-clang))`
`(eval-after-load 'company '(add-to-list 'company-backends 'company-irony-c-headers))`
`(eval-after-load 'company '(add-to-list 'company-backends 'company-irony))`

`(add-hook 'irony-mode-hook 'company-irony-setup-begin-commands)`
(add-hook 'irony-mode-hook 'irony-cdb-autosetup-compile-options)`

In the end the issue was not from company-irony, sorry for the bother
